spec/requests/custom_button_sets_spec.rb

```diff
@@ -0,0 +1,172 @@
+RSpec.describe 'CustomButtonSets API' do
+  let(:cb_set) { FactoryGirl.create(:custom_button_set, :name => 'custom_button_set') }
+  let(:cb_set2) { FactoryGirl.create(:custom_button_set, :name => 'custom_button_set2') }
+
+  describe 'GET /api/custom_button_sets' do
+    it 'does not list custom button sets without an appropriate role' do
+      api_basic_authorize
+
+      get(api_custom_button_sets_url)
+
+      expect(response).to have_http_status(:forbidden)
+    end
+
+    it 'lists all custom button sets with an appropriate role' do
+      api_basic_authorize collection_action_identifier(:custom_button_sets, :read, :get)
+      cb_set_href = api_custom_button_set_url(nil, cb_set)
+
+      get(api_custom_button_sets_url)
+
+      expected = {
+        'count'     => 1,
+        'subcount'  => 1,
+        'name'      => 'custom_button_sets',
+        'resources' => [
+          hash_including('href' => a_string_matching(cb_set_href))
+        ]
+      }
+      expect(response).to have_http_status(:ok)
+      expect(response.parsed_body).to include(expected)
+    end
+  end
+
+  describe 'GET /api/custom_button_sets/:id' do
+    it 'does not let you query custom button sets without an appropriate role' do
+      api_basic_authorize
+
+      get(api_custom_button_set_url(nil, cb_set))
+
+      expect(response).to have_http_status(:forbidden)
+    end
+
+    it 'can query a custom button set by its id' do
+      api_basic_authorize action_identifier(:custom_button_sets, :read, :resource_actions, :get)
+
+      get(api_custom_button_set_url(nil, cb_set))
+
+      expected = {
+        'id'   => cb_set.id.to_s,
+        'name' => cb_set.name
```

Comment:
It can be better to use the literal `"name" => "custom_button_set"` here, as this guarantees you are getting back the thing you expect

Reply:
dd46d57